src/Components/Activities/Form/index.jsx

```diff
@@ -0,0 +1,154 @@
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/${id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+          name,
+          description,
+          isActive
+        }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await res.json();
+      setMessage(data.message);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const onAdd = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/`, {
+        method: 'POST',
+        body: JSON.stringify({
+          name,
+          description,
+          isActive
+        }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await res.json();
+      setMessage(data.message);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleNameChange = (e) => {
+    setName(e.target.value);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    setShowModal(true);
+  };
+
+  const sendActivity = () => {
+    setShowModal(true);
+
+    if (location.pathname.includes('create')) {
+      onAdd({ name, description, isActive });
+      setName('');
+      setDescription('');
+      setIsActive(false);
+      setShowModal(true);
+    }
+
+    if (location.pathname.includes('edit')) {
+      onEdit(id);
+    }
+  };
+
+  return (
+    <div className={styles.formContainer}>
+      <Modal onClose={() => setShowModal(false)} isOpen={showModal} title={message} success />
+      <div className={styles.formBox}>
+        <h3 className={styles.title}>
+          {location.pathname.includes('create') ? 'Add New Activity' : 'Edit Activity'}
+        </h3>
+        <form className={styles.form} onSubmit={(e) => onSubmit(e)}>
+          <div className={styles.field}>
+            <Input
+              labelText={'Name'}
+              type={'text'}
+              placeholder={'Activity name'}
+              value={name}
+              onChangeInput={handleNameChange}
+            />
+          </div>
+          <TextArea
+            label={'Description'}
+            placeholder={'Activity description'}
+            value={description}
+            onChangeArea={setDescription}
+          />
+          <div className={styles.checkboxField}>
+            <label>Status</label>
+            <input
+              className={styles.checkbox}
+              type="checkbox"
+              checked={isActive}
+              value={isActive}
+              onChange={(e) => setIsActive(e.currentTarget.checked)}
+            />
+          </div>
+          <div className={styles.btns}>
+            <Link to="/activities">
+              <Button text={'Cancel'} type={'white'} />
+            </Link>
+            <Button
+              text={location.pathname.includes('edit') ? 'Edit' : 'Add'}
+              type={'add'}
+              clickAction={sendActivity}
```

Comment:
When you create or edit an activity, does it automatically redirect to the list of activities? I can't find the logic for that

Reply:
Im on it!